ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/ImmDB.hs

```diff
@@ -605,49 +605,51 @@ epochFileParser ImmDbArgs{..} =
   where
     -- It is important that we don't first parse all blocks, storing them all
     -- in memory, and only /then/ extract the information we need.
-    decoder' :: forall s. Decoder s (Lazy.ByteString -> (SlotNo, Maybe (HeaderHash blk)))
+    decoder' :: forall s. Decoder s (Lazy.ByteString -> (SlotNo, CurrentEBB (HeaderHash blk)))
     decoder' = (extractSlotNoAndEbbHash .) <$> immDecodeBlock
 
-    extractSlotNoAndEbbHash :: blk -> (SlotNo, Maybe (HeaderHash blk))
+    extractSlotNoAndEbbHash :: blk -> (SlotNo, CurrentEBB (HeaderHash blk))
     extractSlotNoAndEbbHash b =
       -- IMPORTANT: force the slot and the ebbHash, because otherwise we
       -- return thunks that refer to the whole block! See
       -- 'Ouroboros.Consensus.Util.CBOR.readIncrementalOffsets' where we need
       -- to force the decoded value in order to force this computation.
+      --
+      -- TODO: Is this still needed, now that the Maybe is gone..?
```

Comment:
I think so, note the forcing of the slot (otherwise we keep a `blockSlot b` thunk) in addition to the `ebbHash`.

Reply:
Ok, removed.